### PR TITLE
feat(console): configure cors-allowed-origins

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

* Support configure the `corsAllowedOrigins` in the admin console
* Refactor the `MultiTextInput` to support undefined value when configuring the `corsAllowedOrigins`.
* Refactor the `MultiTextInput` to support custom validation logic.
* Updated codes related to the the `MultiTextInput`.
* Refactor the `ApplicationDetails` page for the file is more than 300 lines.
* Align the pattern of all relative URIs. (use `URL` constructor)

### Todo
* LOG-2327 Use Context to Pass the OidcConfig to Settings and AdvancedSettings

## Linear Issue Reference

* LOG-2170

## Testing

image

### CODE REVIEW

File: packages/console/src/components/MultiTextInput/utils.ts [hunk 1]
```diff
@@ -3,27 +3,27 @@ import { conditional } from '@silverhand/essentials';
...
-  if (requiredError) {
-    return {
-      required: requiredError,
-    };
+  if (!value || value.length === 0 || value.filter(Boolean).length === 0) {
+    return conditional(
+      rule.required && {
+        required: rule.required,
+      }
+    );
```
Comment (demonzoo) [inline packages/console/src/components/MultiTextInput/utils.ts:18]: This looks weird... Can it just be something like:
```typescript
if (!value?.filter(Boolean)?.length) {
  return {
    required: rule.required,
  }
}
```

File: packages/console/src/components/MultiTextInput/utils.ts [hunk 2]
```diff
@@ -3,27 +3,27 @@ import { conditional } from '@silverhand/essentials';
...
   }
 
   if (rule.pattern) {
-    const { regex, message } = rule.pattern;
+    const { verify, message } = rule.pattern;
 
     const inputErrors = Object.fromEntries(
       value.map((element, index) => {
-        return [index, regex.test(element) ? undefined : message];
+        return [index, verify(element) ? undefined : message];
```
Comment (demonzoo) [inline packages/console/src/components/MultiTextInput/utils.ts:26]: Why `undefined`? Can we just use [index, true/false] instead?

File: packages/console/src/pages/ApplicationDetails/index.tsx [hunk 3]
```diff
@@ -13,52 +13,47 @@ import Button from '@/components/Button';
...
 import * as modalStyles from '@/scss/modal.module.scss';
 import { applicationTypeI18nKey } from '@/types/applications';
-import { noSpaceRegex } from '@/utilities/regex';
 
+import AdvancedSettings from './components/AdvancedSettings';
 import DeleteForm from './components/DeleteForm';
+import Settings from './components/Settings';
 import * as styles from './index.module.scss';
 
+const mapToUriFormatArrays = (value: string[] | undefined) =>
```
Comment (demonzoo) [inline packages/console/src/pages/ApplicationDetails/index.tsx:32]: value?: string[]

**Comment by demonzoo** [comment]
Please be noted that `CodeComponentRenderer` has been removed from main branch. Please rebase and remove it from your PR as well

File: packages/console/src/components/MultiTextInput/utils.ts [hunk 4]
```diff
@@ -10,12 +8,14 @@ export const validate = (
...
-      }
-    );
+  if (!value?.filter(Boolean).length) {
+    if (!rule.required) {
+      return;
+    }
+
+    return {
+      required: rule.required,
+    };
```
Comment (demonzoo) [inline packages/console/src/components/MultiTextInput/utils.ts:18]: I think it's equivalent to
```typescript
return { required: rule.required }
```

Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/utils.ts]: No, the first one returns undefined, and the second one returns an object.

**Review by demonzoo — APPROVED** [review]

File: packages/console/src/pages/ApplicationDetails/components/Settings.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+          }}
+          render={({ field: { onChange, value }, fieldState: { error } }) => (
+            <MultiTextInput
+              value={value}
+              error={convertRhfErrorMessage(error?.message)}
+              onChange={onChange}
+            />
+          )}
+        />
+      </FormField>
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/pages/ApplicationDetails/components/Settings.tsx:116]: per discussion in slack, refactor the rule instead.
https://silverhand-io.slack.com/archives/C02CMB8JVJB/p1651131172500099

**Pull request merged**